kolibri_zim_plugin/assets/src/views/ZimRendererIndex.vue

```diff
@@ -184,6 +190,21 @@
           this.$refs.zimContentView.navigateToUrl(breadcrumb.href);
         }
       },
+      onZimSearchViewActivate({ path }) {
+        this.isSearching = false;
+        // Assume that the first item in zimNavigationHistory is the home
+        // article, which we want to keep, and remove everything else.
+        this.zimNavigationHistory.splice(1);
+        this.$refs.zimContentView.navigateToArticle(path);
+      },
+      onZimSearchViewCancel() {
+        this.isSearching = false;
+      },
+      onZimSearchOverlayClick(event) {
+        if (event.target === this.$refs.zimSearchOverlay) {
```

Comment:
Why is needed this check here? If I'm understanding the code correctly, this method is linked to the `click` event for the `zimSearchOverlay` div and just for that, so that should always be true and we can remove the `event` and this `if` from the method.

Reply:
That's needed to filter out clicks on an element _inside_ the overlay. Normally those particular events propagate, but we only want the search overlay to close if someone clicks the background directly. If I remove that check, then clicking the form input will cause the search box to close.

Thinking about it, it is kind of funny that Vue doesn't magic that away since it conflicts with the rest of its event model, but I guess it would make it awkward for cases where you actually want `click` to propagate normally…